src/Agent.Listener/DistributedTask.Pipelines/Yaml/TypeConverters/ConverterUtil.phases.cs

```diff
@@ -0,0 +1,515 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters
+{
+    internal static partial class ConverterUtil
+    {
+        internal static IList<IPhase> ReadPhases(IParser parser, Boolean simpleOnly)
+        {
+            var result = new List<IPhase>();
+            parser.Expect<SequenceStart>();
+            while (parser.Allow<SequenceEnd>() == null)
+            {
+                result.Add(ReadPhase(parser, simpleOnly));
+            }
+
+            return result;
+        }
+
+        internal static IPhase ReadPhase(IParser parser, Boolean simpleOnly)
+        {
+            IPhase result;
+            parser.Expect<MappingStart>();
+            Scalar scalar = parser.Expect<Scalar>();
+            if (String.Equals(scalar.Value, YamlConstants.Name, StringComparison.Ordinal))
+            {
+                var phase = new Phase { Name = ReadNonEmptyString(parser) };
+                while (parser.Allow<MappingEnd>() == null)
+                {
+                    scalar = parser.Expect<Scalar>();
+                    switch (scalar.Value ?? String.Empty)
+                    {
+                        case YamlConstants.DependsOn:
+                            if (parser.Accept<Scalar>())
+                            {
+                                scalar = parser.Expect<Scalar>();
+                                if (!String.IsNullOrEmpty(scalar.Value))
+                                {
+                                    phase.DependsOn = new List<String>();
+                                    phase.DependsOn.Add(scalar.Value);
+                                }
+                            }
+                            else
+                            {
+                                phase.DependsOn = ReadSequenceOfString(parser);
+                            }
+
+                            break;
+
+                        case YamlConstants.Condition:
+                            phase.Condition = ReadNonEmptyString(parser);
+                            break;
+
+                        case YamlConstants.ContinueOnError:
+                            phase.ContinueOnError = ReadNonEmptyString(parser);
+                            break;
+
+                        case YamlConstants.EnableAccessToken:
```

Comment:
Seems like we needed this. I need to test when I integrate with VSO repo and make sure it works like I expect (can't test via local run).